flow360/component/flow360_params/params_base.py

```diff
@@ -405,25 +407,61 @@ def _generate_schema_for_optional_objects(cls, model: dict, key: str):
             displayed = field.get("displayed")
 
             if displayed is not None:
-                model["properties"][toggle_name]["displayed"] = displayed
+                schema["properties"][toggle_name]["displayed"] = displayed
 
-            if model.get("dependencies") is None:
-                model["dependencies"] = {}
+            if schema.get("dependencies") is None:
+                schema["dependencies"] = {}
 
-            model["dependencies"][toggle_name] = {
+            schema["dependencies"][toggle_name] = {
                 "oneOf": [
                     {
                         "type": "object",
                         "properties": {
                             toggle_name: {"default": True, "const": True, "type": "boolean"},
-                            key: {"$ref": ref},
+                            key: field,
                         },
                         "additionalProperties": False,
                     },
                     {"additionalProperties": False},
                 ]
             }
 
+    @classmethod
+    def _apply_option_names(cls, dictionary):
+        if not isinstance(dictionary, dict):
+            raise ValueError("Input must be a dictionary")
+
+        for key, value in list(dictionary.items()):
+            options = dictionary.get("anyOf")
+            if key == "options" and options is not None:
+                if len(value) != len(options):
+                    raise ValueError(
+                        f"Tried applying options for {value}, but lengths "
+                        f"of anyOf items and options do not match"
+                    )
+                for option, title in zip(options, value):
+                    option["title"] = title
+            elif isinstance(value, dict):
+                cls._apply_option_names(value)
+            elif isinstance(value, list):
+                for item in value:
+                    if isinstance(item, dict):
+                        cls._apply_option_names(item)
+
+    @classmethod
+    def _fix_single_value_enum(cls, dictionary):
+        for key, value in list(dictionary.items()):
+            if key == "enum" and len(value) == 1:
+                default = value[0]
+                del dictionary[key]
+                dictionary["const"] = default
+            elif isinstance(value, dict):
+                cls._fix_single_value_enum(value)
+            elif isinstance(value, list):
+                for item in value:
+                    if isinstance(item, dict):
+                        cls._fix_single_value_enum(item)
+
```

Comment:
these are exclusively for schema, right? Let's rename functions to reflect it is for schema.

Reply:
Renamed with schema prefixes.